lib/CMakeLists.txt

```diff
@@ -100,7 +100,7 @@ if(WIN32)
     set_target_properties(${LIB_NAME} PROPERTIES IMPORT_SUFFIX "_imp.lib")
 
     set_target_properties (${LIB_NAME} PROPERTIES
-       DEBUG_POSTFIX "-d"
+       DEBUG_POSTFIX "${CMAKE_DEBUG_POSTFIX}"
```

Comment:
The [`CMAKE_<CONFIG>_POSTFIX`](https://cmake.org/cmake/help/v3.11/variable/CMAKE_CONFIG_POSTFIX.html) variable is not meant for direct reference by project code.  Instead it initializes the [`<CONFIG>_POSTFIX`](https://cmake.org/cmake/help/v3.11/prop_tgt/CONFIG_POSTFIX.html) target property set here.  Therefore the change proposed here would be better done by simply removing the property setting altogether.

However, the proposed change will change the default when no `CMAKE_DEBUG_POSTFIX` is set.  Instead one could use `get_property/set_property` to enforce the `-d` default when the property has not been initialized from the variable already.

Reply:
Over in #2121 it is suggested to drop this property setting and instead use

```cmake
if(NOT DEFINED CMAKE_DEBUG_POSTFIX)
  set(CMAKE_DEBUG_POSTFIX "-d")
endif()
```

Reply:
@bradking Thanks. You'll also need:
> along with dropping the explicit DEBUG_POSTFIX property setting should be enough. 

as mentioned in #2121